server/CONTRIBUTING.md

```diff
@@ -112,10 +111,251 @@ pytest --hge-urls http://127.0.0.1:8080 --pg-urls <database_url> -vv
 
 - Compiler warnings are turned on, make sure your code has no warnings.
-- Use [hlint](https://github.com/ndmitchell/hlint) to make sure your code has no warnings.
-- Use [stylish-haskell](https://github.com/jaspervdj/stylish-haskell) to format your code.
+- Use [hlint](https://github.com/ndmitchell/hlint) to make sure your code has no
+  warnings.
+- Use [stylish-haskell](https://github.com/jaspervdj/stylish-haskell) to format
+  your code.
+
+This is a short document describing the preferred coding style for this project.
+We've tried to cover the major areas of formatting and naming. When something
+isn't covered by this guide you should stay consistent with the code in the
+other modules.
+
+### Formatting
+
+#### Line Length
+
+Maximum line length is *80 characters*.
+
+#### Indentation
+
+Tabs are illegal. Use spaces for indenting. Indent your code blocks with *2
+spaces*. Indent the `where` keyword two spaces to set it apart from the rest of
+the code and indent the definitions in a `where` clause 2 spaces. Some examples:
+
+```haskell
+sayHello :: IO ()
+sayHello = do
+  name <- getLine
+  putStrLn $ greeting name
+  where
+    greeting name = "Hello, " ++ name ++ "!"
+
+filter :: (a -> Bool) -> [a] -> [a]
+filter _ []   = []
+filter p (x:xs)
+  | p x       = x : filter p xs
+  | otherwise = filter p xs
+```
+
+#### Blank Lines
+
+One blank line between top-level definitions. No blank lines between type
+signatures and definitions of a function. Add one blank line between functions
+in a type class instance declaration if the function bodies are large. Use your
+judgement.
+
+#### Whitespace
+
+Surround binary operators with a single space on either side. Use your better
+judgement for the insertion of spaces around arithmetic operators but always be
+consistent about whitespace on either side of a binary operator. Don't insert a
+space after a lambda.
+
+#### Data Declarations
+
+Align the constructors in a data type definition. Example:
+
+```haskell
+data Tree a
+  = Branch !a !(Tree a) !(Tree a)
+  | Leaf
+```
+
+```haskell
+data HttpException
+    = InvalidStatusCode Int
+    | MissingContentHeader
+```
+
+Format records as follows:
+
+```haskell
+data Person
+  = Person
+  { firstName :: !String  -- ^ First name
+  , lastName  :: !String  -- ^ Last name
+  , age       :: !Int     -- ^ Age
+  } deriving (Eq, Show)
+```
+
+#### List Declarations
+
+Align the elements in the list.  Example:
+
+```haskell
+exceptions =
+  [ InvalidStatusCode
+  , MissingContentHeader
+  , InternalServerError
+  ]
+```
+
+Optionally, you can skip the first newline.  Use your judgement.
+
+```haskell
+directions = [ North
+             , East
+             , South
+             , West
+             ]
+```
+
+#### Export Lists
+
+Format export lists as follows:
+
+```haskell
+module Data.Set
+  (
+    -- * The @Set@ type
+    Set
+  , empty
+  , singleton
+
+    -- * Querying
+  , member
+  ) where
+```
+
+### Imports
```

Comment:
`stylish-haskell` already manages the import list right? I don't have a strong opinion on this, except that if `stylish-haskell` is to be used it needs to be idempotent for a properly formatted (according to this guide) file. Maybe there's a way to configure it to do what we want?

Reply:
Yeah, I'm not sure. stylish-haskell sorts them alphabetically if there are no newlines in between. We wanted to sort external libs first then hasura libs -> then qualified imports of external libs then hasura libs. I should look if there is stylish haskell configuration to do that. But I don't strongly feel about this, we can probably keep the stylish-haskell defaults.